### DISCUSSION TITLE
How does jaeger authentication work in auto instrumentation .

### DISCUSSION THREAD

**Discussion by vinitaaher** [open-telemetry/opentelemetry-java-instrumentation#2050] [Q&A]
Hi All,
I did auto instrumentation in java for my app and export the traces to jaeger. Following is the parameters I am passing to jvm and it is working.
java -javaagent:tracinglib/opentelemetry-javaagent-all.jar\
-Dotel.exporter=jaeger \
-Dotel.exporter.jaeger.service.name=adService \
-Dotel.exporter.jaeger.endpoint=localhost:14250 \
-jar build/libs/hipstershop-0.1.0-SNAPSHOT-fat.jar 
But now I want do Jaeger authentication using either bearer token or username and password.
Please suggest how to do this.I have bearer token as well username and password also but I don't understand how to pass it with above JVM parameters.

**Comment by anuraaga** [comment]
Hi @vinitaaher - we currently don't support custom authentication of exporters in OpenTelemetry. This is an important feature so I've opened up an issue in the general specification for guidance on how to enable this. Sorry for the trouble

https://github.com/open-telemetry/opentelemetry-specification/issues/1344

**Reply by vinitaaher** [reply]
Thank You @anuraaga

**Reply by umarkhan17** [reply]
Is it available now. they're saying you can implement extensions to enable authentication and pass the bearer token in otel.exporter.otlp.headers. Do we have any reference on how to do it?

**Reply by mateuszrzeszutek** [reply]
It is not available by default as a part of the package - and it won't be until it is present in the OTel specification.
There is a possibility to implement that as a custom javaagent extension, but that your (javaagent user) custom work.
